### PR TITLE
Refactored the consensus components to be used in sovereign chains

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- refactored the consensus components to be used in sovereign chains
  
## Proposed changes
- refactored the consensus components to be used in sovereign chains

## Testing procedure
- N/A

## Pre-requisites

Based on the [Contributing Guidelines](CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

File: consensus/spos/bls/blsSubroundsFactory.go [hunk 1]
```diff
@@ -142,7 +144,7 @@ func (fct *factory) generateStartRoundSubround() error {
 		return err
 	}
 
-	subroundStartRound, err := NewSubroundStartRound(
```
Comment (SebastianMarian) [inline consensus/spos/bls/blsSubroundsFactory.go:145]: Maybe we can keep subround, as a single word, without caps R everywhere in this file

File: consensus/constants.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package consensus
+
+// SubRoundBlockType defines the types of subround block type
+type SubRoundBlockType string
```
Comment (SebastianMarian) [inline consensus/constants.go:4]: Maybe we can keep subround, as a single word, without caps R everywhere in this file

File: consensus/spos/bls/errors.go [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package bls
+
+import "errors"
+
+// ErrUnImplementedSubRoundType signals an un-implemented subround type
+var ErrUnImplementedSubRoundType = errors.New("un-implemented sub-round type")
```
Comment (SebastianMarian) [inline consensus/spos/bls/errors.go:6]: Unimplemented without caps I ?

File: factory/consensus/consensusComponents.go [hunk 4]
```diff
@@ -44,22 +44,23 @@ type ConsensusComponentsFactoryArgs struct {
 	ScheduledProcessor    consensus.ScheduledProcessor
 	IsInImportMode        bool
 	ShouldDisableWatchdog bool
+	SubRoundBlockType     consensus.SubRoundBlockType
```
Comment (SebastianMarian) [inline factory/consensus/consensusComponents.go:47]: Maybe we can keep subround, as a single word, without caps R everywhere in this file

**Pull request merged**